src/Phobos.Kafka/src/Petabridge.Phobos.Kafka.Producer/AkkaService.cs

```diff
@@ -140,4 +234,52 @@ public Task StopAsync(CancellationToken cancellationToken)
             return CoordinatedShutdown.Get(_actors.Sys).Run(CoordinatedShutdown.ClrExitReason.Instance);
         }
     }
+
+    public static class ProducerRecordExtensions
+    {
+        public static ProducerRecord<TKey, TValue> WithTracing<TKey, TValue>(
+            this ProducerRecord<TKey, TValue> record,
+            ActorSystem system,
+            ISpanContext spanContext)
+        {
+            if (spanContext == null)
+                return record;
+            
+            var serializer = system.Serialization.FindSerializerForType(typeof(SpanEnvelope));
+            var envelope = new SpanEnvelope("", spanContext, false);
+            var serialized = serializer.ToBinary(envelope);
+
+            record.Message.Headers ??= new Headers();
+            record.Message.Headers.Add("spanContext", serialized);
+            return record;
+        }
+    }
+    
+    public static class ProducerMessageWithTracing
+    {
+        public static IEnvelope<TKey, TValue, TPassThrough> Single<TKey, TValue, TPassThrough>(
+            ProducerRecord<TKey, TValue> record,
+            ActorSystem system,
+            ISpanContext spanContext,
+            TPassThrough passThrough)
+        {
+            if (spanContext == null)
+                return new Message<TKey, TValue, TPassThrough>(record, passThrough);
+            
+            var serializer = system.Serialization.FindSerializerForType(typeof(SpanEnvelope));
+            var envelope = new SpanEnvelope("", spanContext, false);
```

Comment:
Is this all being done just to serialize the `SpanContext`?

Reply:
Yes, that is used to serialize SpanContext. Should we make a dedicated serializer instead?
To be honest, #17 is a much cleaner implementation than this one.